GEOS-11625: Add "Challenge Anonymous Sessions" Option to AuthKey Filter (#8066)
geoserver/geoserver: doc/en/user/source/extensions/authkey/index.rst

   saves the current configuration and triggers a synchronize. If users are added/removed from 
   the backing user/group service, the synchronize logic should be triggered.

Challenge Anonymous Session
---------------------------

> The AuthKey filter now includes a new option, "Challenge Anonymous Sessions," designed to enforce stricter authorization and stateless behavior.

Reviewer: This is not an announcement... in a few months there will be nothing new about this functionality any more. Best to just state what it does and use present tense

```suggestion
"Challenge Anonymous Sessions" is designed to enforce stricter authorization and stateless behavior.
```

> When enabled, the filter will ignore any existing session or SecurityContext principal (e.g., anonymous or previously authenticated sessions created by other filters) and will always authenticate the user linked to the provided authkey. This ensures that the authkey is validated independently for every request, regardless of the current session state.

Reviewer: ```suggestion
When enabled, the filter ignores any existing session or SecurityContext principal (e.g., anonymous or previously authenticated sessions created by other filters) and always authenticates the user linked to the provided authkey. This ensures that the authkey is validated independently for every request, regardless of the current session state.
```

This feature is particularly useful for deployments that require stateless authentication or need to ensure that requests are processed strictly based on the authkey query parameter. By default, this option is disabled to preserve existing behavior.

**Internal User Cache**

> To enhance performance in stateless mode, an internal user cache has been added to the AuthKey filter. The cache prevents unnecessary backend authentication requests for the same authkey by storing the user information locally for a configurable period.

Reviewer: ```suggestion
To enhance performance in stateless mode, the AuthKey filter can use an internal user cache. The cache prevents unnecessary backend authentication requests for the same authkey by storing the user information locally for a configurable period.
```
